crates/loam-cli/examples/soroban/nft/src/nft.rs

```diff
@@ -0,0 +1,126 @@
+use loam_sdk::{
+    soroban_sdk::{self, contracttype, env, Address, Bytes, Lazy, Map, Vec},
+    IntoKey,
+};
+use loam_subcontract_core::Core;
+
+use crate::{
+    subcontract::{IsInitable, IsNonFungible},
+    Contract,
+};
+
+#[contracttype]
+#[derive(IntoKey)]
+pub struct MyNonFungibleToken {
+    admin: Address,
+    name: Bytes,
+    total_count: u32,
+    owners_to_nft_ids: Map<Address, Map<u32, ()>>, // the owner's collection
+    nft_ids_to_owners: Map<u32, Address>,
+    nft_ids_to_metadata: Map<u32, Bytes>,
+}
+
+impl MyNonFungibleToken {
+    #[must_use]
+    pub fn new(admin: Address, name: Bytes) -> Self {
+        MyNonFungibleToken {
+            admin,
+            name,
+            total_count: 0,
+            owners_to_nft_ids: Map::new(env()),
+            nft_ids_to_owners: Map::new(env()),
+            nft_ids_to_metadata: Map::new(env()),
+        }
+    }
+}
+
+impl Default for MyNonFungibleToken {
+    fn default() -> Self {
+        MyNonFungibleToken::new(env().current_contract_address(), Bytes::new(env()))
+    }
+}
+
+impl IsInitable for MyNonFungibleToken {
+    fn nft_init(&self, admin: Address, name: Bytes) {
+        Contract::admin_get().unwrap().require_auth();
+        MyNonFungibleToken::set_lazy(MyNonFungibleToken::new(admin, name));
+    }
+}
+
+impl IsNonFungible for MyNonFungibleToken {
+    // Mint a new NFT with the given owner address and metadata, returning the id
```

Comment:
I wonder if we should model copying the comments from the subcontract into the implementation, or leave the comments out in this file. The ones in the `#[subcontract]` are the ones that show up in contract calls.

Reply:
Great point - we can probably leave them out of this file and rely on the ones in the subcontract file 👍

Reply:
[ba7af15](https://github.com/loambuild/loam-sdk/pull/91/commits/ba7af1509a108903b64a29721fcf2f6068cc3ec3)